### ISSUE TITLE
Provide explanations on used formatting strings for custom date/time format

### ISSUE THREAD

**Issue by Porkepix** [martykan/forecastie#437]
Currently it's possible to set a custom date/time format, but there's not hint as to which formatter to use for it.

**Comment by robinpaulson** [comment]
I think it's something called "Simple date format", a search on the web will find it. That said, we should perhaps give a better idea of how to use it.
Here's some docs, written for programmers, not end users:
https://docs.oracle.com/javase/7/docs/api/java/text/SimpleDateFormat.html

**Comment by robinpaulson** [comment]
I recall we talked about this previously. Offering freeform examples is good... up to a point. Maybe we only offer a fixed set of formats, the user can then choose one.

**Comment by Porkepix** [comment]
Problem with this is that you can't satisfy anyone with completely set proposals, or you allow to pick separately setting for different elements, eg. separators, number order for date and order between day, date and time.

For example my chosen format would be dd/mm/yyyy hh:mm which isn't covered by the presets. And other people would do different choices.

**Comment by robinpaulson** [comment]
There are a few bugs related to this:
#326 #287 #219 #272 #361

**Comment by robinpaulson** [comment]
> And other people would do different choices.

At some point, the code to support those choices becomes more complicated than I think is necessary. Sure, this is to a certain extent arbitrary, but it is a relatively minor setting. As was pointed out in one of the bugs I listed above, there are likely libraries for this, are you prepared to put time into coding something?

**Comment by Porkepix** [comment]
> As was pointed out in one of the bugs I listed above, there are likely libraries for this, are you prepared to put time into coding something?

Sadly my skills are more about taking care of servers than code itself, and about code especially not Java :/

**Comment by jawz101** [comment]
I have seen a few places where they have a tiny legend which helps me experiment with the choices.  It may show all potiential characters, a popular long variation, and then what it represents.  That's enough for me to figure out what I want.

If the popup included these 3 lines it would give me enough of a "decoder" to figure out what I can do

```
GyYMwWDdFEuaHkKhmsSzZX
EEEE, d MMM yyyy h:mm:ss a, zzzz
Wednesday, 4 Jul 2001 12:08:32 PM, Pacific Daylight Time
```

**Comment by robinpaulson** [comment]
> it would give me enough of a "decoder" to figure out what I can do

True, it would. It likely looks intimidating and overwhelming to most users though, I'm not keen on putting things like that in the main interface, I'd prefer them at least behind a button or an "Advanced" tab.

**Comment by jawz101** [comment]
Oh, definitely.  Like in the popup

**Comment by robinpaulson** [comment]
We'll need to add some more strings to make this work; can you give me some suggestions as to how to name the strings you posted in your message from Feb 3rd? These:
yYMwWDdFEuaHkKhmsSzZX

EEEE, d MMM yyyy h:mm:ss a, zzzz

Wednesday, 4 Jul 2001 12:08:32 PM, Pacific Daylight Time

**Comment by robinpaulson** [comment]
I think these cover it, I'll add them to strings.xml later today

simple_string_formatting_intro
available_simple_strings
example_simple_strings
example_date_format

**Comment by robinpaulson** [comment]
> Oh, definitely. Like in the popup

I've just realised: the only way a user will see the codes is if they have (a) selected custom codes and (b) clicked the button to enter a custom code. It's probably overkill to then put the codes in *another* dialogue, so I'm going to put them directly in the "Enter custom code" dialogue, not behind another button.
